tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright 2023 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::convert::TryFrom;
+use tss_esapi::{
+    constants::tss::{
+        TPM2_RC_1, TPM2_RC_2, TPM2_RC_3, TPM2_RC_4, TPM2_RC_5, TPM2_RC_6, TPM2_RC_7, TPM2_RC_8,
+        TPM2_RC_9, TPM2_RC_A, TPM2_RC_B, TPM2_RC_C, TPM2_RC_D, TPM2_RC_E, TPM2_RC_F, TPM2_RC_H,
+        TPM2_RC_P, TPM2_RC_S,
+    },
+    error::ArgumentNumber,
+};
+
+const ARGUMENT_NUMBERS: [u16; 15] = [
+    TPM2_RC_1 as u16,
+    TPM2_RC_2 as u16,
+    TPM2_RC_3 as u16,
+    TPM2_RC_4 as u16,
+    TPM2_RC_5 as u16,
+    TPM2_RC_6 as u16,
+    TPM2_RC_7 as u16,
+    TPM2_RC_8 as u16,
+    TPM2_RC_9 as u16,
+    TPM2_RC_A as u16,
+    TPM2_RC_B as u16,
+    TPM2_RC_C as u16,
+    TPM2_RC_D as u16,
+    TPM2_RC_E as u16,
+    TPM2_RC_F as u16,
+];
+
+#[test]
+fn test_valid_parameter_conversions() {
+    let parameter_values = (1u8..16u8).collect::<Vec<u8>>();
+
+    // Test Parameters
+    ARGUMENT_NUMBERS.iter().zip(parameter_values).for_each(|(&nr, val)| {
+        // Ex:
+        // TPM2_RC_1 || TPM_RC_P = 0x140
+        // 0 0 0 1    0 1 |0 0    0 0 0 0
+        //              P |---Error nr---|
+        // Only bits 6 to 11 is parsed by ArgumentNumber.
+        // P bit is set, indicating parameter argument.
+        let return_code_argument_info: u16 = nr | (TPM2_RC_P as u16);
+        let actual = u8::try_from(
+            return_code_argument_info
+                .checked_shr(6).expect("Failed to extract argument information.")
+            )
+            .map (ArgumentNumber::from)
+            .unwrap_or_else(|_| panic!("Failed to convert argument information {:#b} to u8.", return_code_argument_info));
```

Comment:
:thinking: I'm wondering if `.unwrap_or_else(|_| panic!(` can just be folded into `expect` here?

Reply:
If you write ```.expect(format!())``` you get a lint error. https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call

Reply:
Ah, right. Thanks for the link and the explanation! 👋